WordPress/gutenberg
# Tooltip: Fix positioning by anchoring to child element

## What?


Alternative to #41176 

This PR updates the ToolTip component to ensure that its position is correctly relative to the first child of the ToolTip.



## Why?


Following on from #40740 where the Popover component was refactored to use the floatingUI library, it looks like the following issues were introduced:

* Tooltips were no longer centred relative to their child component
* Tooltips at the edge of the screen would overflow / get a horizontal scrollbar

It looks like it was also mentioned in this comment: https://github.com/WordPress/gutenberg/pull/40740#pullrequestreview-966004363

## How?


* Add a ref to the ToolTip component that gets passed to both the ToolTip's child, and as an anchorRef, used as a reference for the Popover — this ensures that the Popover has the correct reference point for calculating the Tooltip position.

* Set the min-width of the Tooltip to `min-content` instead of `0` to prevent horizontal scrollbars from appearing

## Testing Instructions





1. Test in Storybook and ensure that by default the `ToolTip` component is now positioned relative to its first child.
2. In the post editor, hover over the options (vertical ellipsis) at the top right, and check that the Tooltip does not overflow / introduce a scrollbar.
3. Hover over the Undo / Redo buttons and ensure the Tooltip is now positioned more closely to the expected position.
4. Insert a block, and hover over the controls in the block toolbar: the tooltips should now be centered to the button.

## Screenshots or screencast 

| Before | After |
| --- | --- |
| <img width="69" alt="image" src="https://user-images.githubusercontent.com/14988353/169954716-546b5c09-caf9-4e0f-8203-73312d183b6d.png"> | <img width="77" alt="image" src="https://user-images.githubusercontent.com/14988353/169954451-a381cc7f-5d22-4f47-b2b4-c5b5196f30ab.png"> |
| <img width="200" src="https://user-images.githubusercontent.com/14988353/169954504-77fb8ec0-7a72-4a3d-8d7c-343a7a609e2d.png" /> | <img width="196" alt="image" src="https://user-images.githubusercontent.com/14988353/169954561-e846ec8e-1864-4417-9d2a-05d5164b152e.png"> |
| <img width="115" alt="image" src="https://user-images.githubusercontent.com/14988353/169954801-c939ff42-382e-4fea-9799-29e44e45a3a3.png"> | <img width="106" alt="image" src="https://user-images.githubusercontent.com/14988353/169954858-51957d2d-886e-4791-a751-1683dfad9348.png"> |
| <img width="300" alt="image" src="https://user-images.githubusercontent.com/14988353/169955130-8a2b5628-a9f9-45cc-b02a-e3d24eb53b77.png"> | <img width="293" alt="image" src="https://user-images.githubusercontent.com/14988353/169955023-00a6b019-bc15-464a-9cb5-c70442378dbb.png"> |
